src/unlock/omni_lock.rs

```diff
@@ -0,0 +1,279 @@
+};
+pub const IDENTITY_FLAGS_PUBKEY_HASH: u8 = 0;
+
+#[derive(Clone, Serialize, Deserialize)]
+pub struct Identity {
+    pub flags: u8,
+    pub blake160: Bytes,
+}
+impl Identity {
+    pub fn to_smt_key(&self) -> [u8; 32] {
+        let mut ret: [u8; 32] = Default::default();
+        ret[0] = self.flags;
+        (&mut ret[1..21]).copy_from_slice(self.blake160.as_ref());
+        ret
+    }
+}
+
+impl From<Identity> for [u8; 21] {
+    fn from(id: Identity) -> Self {
+        let mut res = [0u8; 21];
+        res[0] = id.flags;
+        res[1..].copy_from_slice(&id.blake160);
+        res
+    }
+}
+
+impl From<Identity> for Vec<u8> {
+    fn from(id: Identity) -> Self {
+        let mut bytes = vec![id.flags];
+        bytes.extend(id.blake160.clone());
+        bytes
+    }
+}
+
+impl From<Identity> for ckb_types::bytes::Bytes {
+    fn from(id: Identity) -> Self {
+        let v: Vec<u8> = id.into();
+        v.into()
+    }
+}
+
+impl Display for Identity {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let v: Vec<u8> = self.clone().into();
+        for i in v {
+            write!(f, "{:02x}", i)?;
+        }
+        Ok(())
+    }
+}
+
+/*
+<1 byte Omnilock flags>
+<32 byte RC cell type ID, optional>
+<2 bytes minimum ckb/udt in ACP, optional>
+<8 bytes since for time lock, optional>
+<32 bytes type script hash for supply, optional>
+*/
+
+#[derive(Clone, Serialize, Deserialize)]
+pub struct OmniLockConfig {
+    pub type_hash: H256,
+    pub id: Identity,
+    pub omni_lock_flags: u8,
+}
+
+impl OmniLockConfig {
+    pub fn new_pubkey_hash_with_lockarg(type_hash: H256, lock_arg: Bytes) -> Self {
+        Self::new(type_hash, IDENTITY_FLAGS_PUBKEY_HASH, lock_arg)
+    }
+
+    pub fn new_pubkey_hash(type_hash: H256, pubkey: &Pubkey) -> Self {
+        let pubkey_hash = blake160(&pubkey.serialize());
+        Self::new(type_hash, IDENTITY_FLAGS_PUBKEY_HASH, pubkey_hash)
+    }
+
+
+    pub fn new(type_hash: H256, flags: u8, blake160: Bytes) -> Self {
+        let blake160 = if flags == IDENTITY_FLAGS_PUBKEY_HASH {
+            assert!(blake160.len() == 20);
+            blake160
+        } else {
+            Bytes::from(&[0; 20][..])
+        };
+
+        OmniLockConfig {
+            type_hash,
+            id: Identity { flags, blake160 },
+            omni_lock_flags: 0,
+        }
+    }
+
+    pub fn build_args(&self) -> Bytes {
+        let mut bytes = BytesMut::with_capacity(128);
+
+        // auth
+        bytes.put_u8(self.id.flags);
+        bytes.put(self.id.blake160.as_ref());
+        bytes.put_u8(self.omni_lock_flags);
+
+        bytes.freeze()
+    }
+
+    pub fn is_pubkey_hash(&self) -> bool {
+        self.id.flags == IDENTITY_FLAGS_PUBKEY_HASH
+    }
+
+    pub fn to_address_payload(&self) -> AddressPayload {
+        let args = self.build_args();
+        AddressPayload::new_full(ScriptHashType::Type, self.type_hash.pack(), args)
+    }
```

Comment:
Since `omni-lock` is not a [system script](https://github.com/nervosnetwork/ckb-system-scripts/tree/master/c) this function can not implement in `ckb-sdk-rust` it's for upper level to implement. And should also remove `type_hash` field.